test/jdk/sun/security/validator/CertReplace.java

```diff
@@ -26,22 +26,190 @@
+ * @library /test/lib
+ * @modules java.base/sun.security.validator
+ *
+ * @run main/othervm CertReplace certreplace.jks certreplace.certs
+ */
+
+/*
+ * @test id=samedn
+ * @bug 6958869
+ * @summary Regression: PKIXValidator fails when multiple trust anchors have same dn
+ * @library /test/lib
+ * @modules java.base/sun.security.validator
+ *
+ * @run main/othervm CertReplace samedn.jks samedn1.certs
+ * @run main/othervm CertReplace samedn.jks samedn2.certs
+ */
+
 public class CertReplace {
 
+    private static final String SAMEDN_JKS = "samedn.jks";
+    private static final String CERTREPLACE_JKS = "certreplace.jks";
+    private static final String PASSWORD = "changeit";
+
+    /**
+     * This method creates certs for the Cert Replace test
+     *
+     * @throws Exception
+     */
+    private static void certReplace() throws Exception {
+
+        final String intAliase = "int";
+        final String userAlias = "user";
+        final String caAlias = "ca";
+
+        final String certplaceCerts = "certreplace.certs";
+
+        final String ktBaseParameters = "-storepass " + PASSWORD + " " +
+                                        "-keypass " + PASSWORD + " " +
+                                        "-keystore " + CERTREPLACE_JKS + " " +
+                                        "-keyalg rsa ";
+
+        final Path keystoreFilePath = Paths.get(CERTREPLACE_JKS);
+        Files.deleteIfExists(keystoreFilePath);
+
+        // 1. Generate 3 aliases in a keystore: ca, int, user
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias " + caAlias + " -dname CN=CA -keyalg rsa -sigalg md2withrsa -ext bc");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias " + intAliase + " -dname CN=Int -keyalg rsa");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias " + userAlias + " -dname CN=User -keyalg rsa");
+
+        final KeyStore keyStore = KeyStoreUtils.loadKeyStore(CERTREPLACE_JKS, PASSWORD);
+
+        // 2. Signing: ca -> int -> user
+        final String intReqFile = intAliase + ".req";
+        final String intCertFileName = intAliase + ".cert";
+
+        SecurityTools.keytool(ktBaseParameters +
+                              "-certreq -alias " + intAliase + " -file " + intReqFile);
+        SecurityTools.keytool(ktBaseParameters +
+                              "-gencert -rfc -alias " + caAlias + " -ext bc -infile " + intReqFile + " " +
+                              "-outfile " + intCertFileName);
+
+        //putting the certificate in the keystore
+        final CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+        final Certificate[] certs = new Certificate[]{
+                certificateFactory.generateCertificate(
+                        new FileInputStream(intCertFileName)
+                )
+        };
```

Comment:
Use `CertUtils.getCertFromStream`. Also, it's better to close the file. Maybe Windows dislikes open files.